### PR TITLE
[FLINK-1483] IOManager puts temp files in dedicated directory and cleans up on shutdown

### PR DESCRIPTION
**Pull request by StephanEwen**

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/io/disk/iomanager/IOManager.java [hunk 1]
```diff
@@ -52,26 +57,88 @@
...
 	/**
 	 * Utility method to check whether the IO manager has been properly shut down.
+	 * For this base implementation, this means that all files have been removed.
 	 *
 	 * @return True, if the IO manager has properly shut down, false otherwise.
 	 */
-	public abstract boolean isProperlyShutDown();
+	public boolean isProperlyShutDown() {
+		for (File path : paths) {
+			if (path != null) {
```
Comment (hsaputra) [inline flink-runtime/src/main/java/org/apache/flink/runtime/io/disk/iomanager/IOManager.java]: Would this easier to read with check  {{if(path != null && path.exists())}}

**Comment by hsaputra** [comment]
Since the IOManager add shutdown hook to clean up the files, should IOManagerAsync#isProperlyShutDown need to call super.isProperlyShutDown ?

**Comment by StephanEwen** [comment]
Good comments. I address them and merge this.

The tests that fail are not related to this change (there is an instability in the TaskManager tests currently)

**Comment by StephanEwen** [comment]
Actually, `IOManagerAsync#isProperlyShutDown` already calls the superclass method:
```
return isShutdown.get() && readersShutDown && writersShutDown && super.isProperlyShutDown();
```

**Pull request closed without merging**